pkg/dwarf/line/line_parser.go

```diff
@@ -307,16 +308,20 @@ func parseFileEntries5(info *DebugLineInfo, buf *bytes.Buffer) bool {
 				// not implemented
 			}
 
-			if info.normalizeBackslash {
-				p = strings.ReplaceAll(p, "\\", "/")
+			if p == "" {
```

Comment:
I'm thinking that this is still wrong. Compilers are free to order the contents as:

    DW_LNCT_path DW_LNCT_directory_index

and
    
    DW_LNCT_directory_index DW_LNCT_path

This works in the first case because p will be "" and diridx will be correct, when this loop is examining the DW_LNCT_directory_index entry. It won't work in the second case because we don't save the directory index anywhere.

I think that the code that was at the end of the `for fileEntryFormReader.next(buf)`, starting with the `if info.normalizeBackslash` statement, should be moved outside of the loop, which will guarantee that we add exactly `fileCount` entries to the table (consequently the variables `p` and `diridx` also need to be oustide of that loop).

Reply:
Ack, good call. Done.